File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0 , right = height.length-1;
    let max = 0;
    while(left<=right){
        let val = Math.min(height[left],height[right])*(right-left);
        max = Math.max(max,val);
        if(height[left]<height[right]) left++;
        else right--;
    }
    return max;
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length-1, max =0;
        while(left<right){
            int sum = Math.min(height[left],height[right])*(right-left);
            max = Math.max(sum, max);
            if(height[left]>height[right]){
                right--;
            }else left++;
        }

        return max;
    }
}